### PR TITLE
[SHELL32] Map Alt+... menu keys

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

On English Explorer, `Alt+F` will open the `"&File"` menu. `Alt+E` will open the `"&Edit"` menu. `Alt+V` will open the `"&View"` menu. etc.

JIRA issue: [CORE-12533](https://jira.reactos.org/browse/CORE-12533)

## Proposed changes

- Catch keys at `GetMsgHook`.
- Pass keys into `KeyboardItemChange`.
- Use `TB_MAPACCELERATOR` message in `CMenuToolbarBase::KeyboardItemChange`.

## TODO

- [x] Do tests.
- [ ] Correct submenu navigation.

## Movie

https://github.com/reactos/reactos/assets/2107452/884b35af-fabc-4326-9186-3313f5c6e101
Some bugs remain. Done is better than perfect.

### CODE REVIEW

**Comment by JoachimHenze** [comment]
> Alt+F will open the "&File" menu

What does it do on German Localization? The file menu is supposed to open on "Alt + D", that's what Windows does, and that is what our current accelerators in the de-DE.rc do imply as well.

**Comment by katahiromz** [comment]
> What does it do on German Localization?
`TB_MAPACCELERATOR` will also handle that case correctly.

**Review by JoachimHenze — APPROVED** [review]
Approved. I do recommend waiting for a 2nd approval though.

File: dll/win32/shell32/shellmenu/CMenuFocusManager.cpp
Comment (JoachimHenze) [inline dll/win32/shell32/shellmenu/CMenuFocusManager.cpp:665]:
```suggestion
                     ('0' <= msg->wParam && msg->wParam <= '9')))
```
Do we miss a closing brace here? 3 levels of opening, 2 levels of closing as far as I can see/count.

Comment (JoachimHenze) [reply dll/win32/shell32/shellmenu/CMenuFocusManager.cpp]: resolved

File: dll/win32/shell32/shellmenu/CMenuBand.cpp
Comment (HBelusca) [inline dll/win32/shell32/shellmenu/CMenuBand.cpp:933]: This else-block looks like the VK_UP || VK_END case from above. Is it to be expected?
If so, would it be better to move the middle-block VK_DOWN || VK_HOME first, and make the VK_UP || VK_END the default?

Maybe changing this to a switch could be better?
```c
        switch (change)
        {
        case VK_DOWN: case VK_HOME:
            if (m_SFToolbar)
                tb = m_SFToolbar;
            else
                tb = m_static Toolbar;
            break;

        case VK_UP: case VK_END:
        default:
            if (m_staticToolbar)
                tb = m_staticToolbar;
            else
                tb = m_SFToolbar;
            break;
        }
```
----
I personally wonder why the if-tests for the toolbars are done the way they are: why sometimes it's the `m_ staticToolbar` that is being checked, and other times it's `m_SFToolbar` instead.

File: dll/win32/shell32/shellmenu/CMenuBand.cpp
Comment (HBelusca) [inline dll/win32/shell32/shellmenu/CMenuBand.cpp:996]: Nice bug fix.

**Comment by katahiromz** [comment]
I can't get correct implementation. It's beyond my ability. Closed.

**Pull request closed without merging**
